webapp/channels/src/packages/mattermost-redux/src/types/extend_redux.d.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// This is copied from an older version of redux-thunk
+
+/* eslint-disable */
+
+import { Action, ActionCreatorsMapObject, UnknownAction } from 'redux'
+import { ThunkAction } from 'redux-thunk'
+
+/**
+ * Globally alter the Redux `bindActionCreators` and `Dispatch` types to assume
+ * that the thunk middleware always exists, for ease of use.
+ * This is kept as a separate file that may be optionally imported, to
+ * avoid polluting the default types in case the thunk middleware is _not_
+ * actually being used.
+ *
+ * To add these types to your app:
+ * import 'redux-thunk/extend-redux'
+ */
+declare module 'redux' {
+  /**
+   * Overload for bindActionCreators redux function, returns expects responses
+   * from thunk actions
+   */
+  function bindActionCreators<
+    ActionCreators extends ActionCreatorsMapObject<any>
+  >(
+    actionCreators: ActionCreators,
+    dispatch: Dispatch
+  ): {
+    [ActionCreatorName in keyof ActionCreators]: ReturnType<
+      ActionCreators[ActionCreatorName]
+    > extends ThunkAction<any, any, any, any>
```

Comment:
Can you elaborate a bit more on what these types do? I'm seeing a lot of `any` here so I'm just wondering how they help us.